Update docker-py

yongtang opened the pull request:
This fix update docker-py so that containers from the tests run
could be cleaned up during teardown:
```diff
-ENV DOCKER_PY_COMMIT 5e28dcaace5f7b70cbe44c313b7a3b288fa38916
+ENV DOCKER_PY_COMMIT 8b246db271a85d6541dc458838627e89c683e42f
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
This PR depends on https://github.com/docker/docker-py/pull/1922

thaJeztah commented on Dockerfile:
```diff
@@ -142,9 +142,9 @@ RUN set -x \
+ENV DOCKER_PY_COMMIT dad7ae12d717d7a4bfd003f4d448d1c2805de039
 # To run integration tests docker-pycreds is required.
-RUN git clone https://github.com/docker/docker-py.git /docker-py \
+RUN git clone https://github.com/yongtang/docker-py.git /docker-py \
```
^^ commenting just so that we don't forget to change it before merging

thaJeztah commented:
https://github.com/docker/docker-py/pull/1922 was merged 👍

yongtang committed:
Update docker-py

This fix update docker-py so that containers from the tests run
could be cleaned up during teardown:
```diff
-ENV DOCKER_PY_COMMIT 5e28dcaace5f7b70cbe44c313b7a3b288fa38916
+ENV DOCKER_PY_COMMIT 8b246db271a85d6541dc458838627e89c683e42f
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @thaJeztah. The PR has been updated.

thaJeztah reviewed (approved):
LGTM (if green)

thaJeztah commented:
ping @vdemeester @cpuguy83 it's green 🤗

boaz0 reviewed (approved):
LGTM :+1:

vdemeester reviewed (approved):
LGTM 🦌
